### ISSUE TITLE
Add support for interfaces in cfscript

### ISSUE THREAD

**Issue by infernocloud** [atuttle/atom-language-cfml#84]
A little known feature of Coldfusion is that you can actually create component interfaces. The CFScript version of interfaces is very similar to components but with a few key differences. Currently the atom-language-cfml package does not correctly highlight CFScript interfaces.

Here is an example interface:

```
interface {
	public void function updateObject(required string bucket, required string fileKey);
	public void function loadObject(numeric rangeMin=0, numeric rangeMax=0);
	public any function getObjectInputStream(numeric rangeMin=0, numeric rangeMax=0);
	public string function getObjectURL(numeric lifetime=5);
	public boolean function objectExists();
	public struct function getObjectMetadata();
}
```

Interface function definitions don't include curly brackets and they end in a semi-colon.

Here is an example implementation:
```
component implements="CloudStorageProvider" {
	public GoogleCloudObject function init(required string bucket, required string fileKey) {
		//Functionality here
		return this;
	}

	public void function updateObject(required string bucket, required string fileKey) {
		//Functionality here
	}

	public void function loadObject(numeric rangeMin=0, numeric rangeMax=0) {
		//Functionality here
	}

	public any function getObjectInputStream(numeric rangeMin=0, numeric rangeMax=0) {
		//Functionality here
	}

	public string function getObjectURL(numeric lifetime=5) {
		//Functionality here
	}

	public boolean function objectExists() {
		//Functionality here
	}

	public struct function getObjectMetadata() {
		//Functionality here
	}
}
```

An odd quirk is that functions in an interface can have public access only. Declaring an interface function as private or implementing a function and changing it to private will cause an error.

**Comment by infernocloud** [comment]
This is a feature request with a "Please :)" of course. I can't add labels to issues it seems.

**Comment by atuttle** [comment]
Should be resolved by #87 so I'm going to go ahead and close this. If it's not, please comment and/or reopen as necessary.
